test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -0,0 +1,1473 @@
+        ctrl @ cx c1, c2, a;
+    }
+
+    qubit q1;
+    qubit q2;
+    qubit q3;
+    qubit q4;
+    qubit q5;
+
+    // doesn't flip q2
+    cx q1, q2;
+    // flip q1
+    x q1;
+    // flip q2
+    cx q1, q2;
+    // doesn't flip q3, q4, q5
+    ccx_1 q1, q4, q3;
+    ccx_2 q1, q3, q4;
+    ccx_3 q1, q3, q5;
+    // flip q3, q4, q5;
+    ccx_1 q1, q2, q3;
+    ccx_2 q1, q2, q4;
+    ccx_2 q1, q2, q5;
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    simulation = StateVectorSimulation(5, 1, 1)
+    simulation.evolve(circuit.instructions)
+    assert np.allclose(simulation.state_vector, np.array([0] * 31 + [1]))
+
+
+def test_gate_ctrl_global():
+    qasm = """
+    qubit q1;
+    qubit q2;
+
+    h q1;
+    h q2;
+    ctrl @ s q1, q2;
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    simulation = StateVectorSimulation(2, 1, 1)
+    simulation.evolve(circuit.instructions)
+    assert np.allclose(
+        simulation.state_vector,
+        [0.5, 0.5, 0.5, 0.5j],
+    )
+
+
+def test_neg_gate_ctrl():
+    qasm = """
+    int[8] two = 2;
+    gate x a { U(π, 0, π) a; }
+    gate cx c, a {
+        negctrl @ x c, a;
+    }
+    gate ccx_1 c1, c2, a {
+        negctrl @ negctrl @ x c1, c2, a;
+    }
+    gate ccx_2 c1, c2, a {
+        negctrl(two) @ x c1, c2, a;
+    }
+    gate ccx_3 c1, c2, a {
+        negctrl @ cx c1, c2, a;
+    }
+
+    qubit q1;
+    qubit q2;
+    qubit q3;
+    qubit q4;
+    qubit q5;
+
+    x q1;
+    x q2;
+    x q3;
+    x q4;
+    x q5;
+
+    // doesn't flip q2
+    cx q1, q2;
+    // flip q1
+    x q1;
+    // flip q2
+    cx q1, q2;
+    // doesn't flip q3, q4, q5
+    ccx_1 q1, q4, q3;
+    ccx_2 q1, q3, q4;
+    ccx_3 q1, q3, q5;
+    // flip q3, q4, q5;
+    ccx_1 q1, q2, q3;
+    ccx_2 q1, q2, q4;
+    ccx_3 q1, q2, q5;
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    simulation = StateVectorSimulation(5, 1, 1)
+    simulation.evolve(circuit.instructions)
+    assert np.allclose(simulation.state_vector, np.array([1] + [0] * 31))
+
+
+def test_pow():
+    qasm = """
+    int[8] two = 2;
+    gate x a { U(π, 0, π) a; }
+    gate cx c, a {
+        pow(1) @ ctrl @ x c, a;
+    }
+    gate cxx_1 c, a {
+        pow(two) @ cx c, a;
+    }
+    gate cxx_2 c, a {
+        pow(1/2) @ pow(4) @ cx c, a;
+    }
+    gate cxxx c, a {
+        pow(1) @ pow(two) @ cx c, a;
+    }
+
+    qubit q1;
+    qubit q2;
+    qubit q3;
+    qubit q4;
+    qubit q5;
+
+    x q1;       // flip
+    cx q1, q2;   // flip
+    cxx_1 q1, q3;    // don't flip
+    cxx_2 q1, q4;    // don't flip
+    cnot q1, q5;    // flip
+    x q3;       // flip
+    x q4;       // flip
+
+    s q1;   // sqrt z
+    s q1;   // again
+    inv @ z q1; // inv z
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    simulation = StateVectorSimulation(5, 1, 1)
+    simulation.evolve(circuit.instructions)
+    assert np.allclose(simulation.state_vector, np.array([0] * 31 + [1]))
+
+
+def test_measurement():
```

Comment:
what is this testing?

Reply:
We expect to support more robust outputs using the measure instruction in the future, so this will be more fleshed out then